ddtrace/contrib/celery/app.py

```diff
@@ -26,8 +35,19 @@ def patch_app(app, pin=None):
         _config=config.celery,
     )
     pin.onto(app)
-    # connect to the Signal framework
 
+    wrapt.wrap_function_wrapper(
+        "celery.beat",
+        "Scheduler.apply_entry",
+        _traced_beat_function(config.celery, "apply_entry", lambda args: args[0].name),
+    )
+    # If celery.Celery.conf.broker_max_connection_retries is not set and
+    # celery.beat.Scheduler's connection_for_writing is unavailable at startup, _ensure_connected will spin
+    # forever and this integration will not generate any traces until that connection becomes available.
```

Comment:
Is this an implementation detail on the user-side that we can't check? I.e. is there no way for us to detect `celery.Celery.conf.broker_max_connection_retries` not being set and `celery.beat.Scheduler.connection_for_writing` unavailable at startup, and warn the user accordingly?

Reply:
Revisiting this with a few days of distance, I realize this comment isn't necessary. The "spinning" condition it describes is something completely internal to celery beat and not related to the tracer's presence in the runtime.